src/fromager/wheels.py

```diff
@@ -129,6 +131,73 @@ def analyze_wheel_elfdeps(
     return requires, provides
 
 
+def add_extra_metadata_to_wheels(
+    ctx: context.WorkContext,
+    req: Requirement,
+    version: Version,
+    extra_environ: dict[str, str],
+    sdist_root_dir: pathlib.Path,
+    wheel_file: pathlib.Path,
+) -> None:
+    extra_data_plugin = overrides.find_override_method(
+        req.name, "add_extra_metadata_to_wheels"
+    )
+    data_to_add = {}
+    if extra_data_plugin:
+        data_to_add = overrides.invoke(
+            extra_data_plugin,
+            ctx=ctx,
+            req=req,
+            version=version,
+            extra_environ=extra_environ,
+            sdist_root_dir=sdist_root_dir,
+        )
+        if not isinstance(data_to_add, dict):
+            logger.warning(
+                f"{req.name}: unexpected return type from plugin add_extra_metadata_to_wheels. Expected dictionary. Will ignore"
+            )
+            data_to_add = {}
+
+    with tempfile.TemporaryDirectory() as dir_name:
+        cmd = ["wheel", "unpack", str(wheel_file), "-d", dir_name]
+        external_commands.run(
+            cmd,
+            cwd=dir_name,
+            network_isolation=ctx.network_isolation,
+        )
+
+        wheel_file_name_parts = wheel_file.name.split("-")
+        distribution_name = f"{wheel_file_name_parts[0]}-{wheel_file_name_parts[1]}"
+        dist_info_dir = (
+            pathlib.Path(dir_name)
+            / distribution_name
+            / f"{distribution_name}.dist-info"
+        )
+        if not dist_info_dir.exists():
+            logger.debug(
+                f"{req.name}: could not add additional metadata. {dist_info_dir} does not exist"
+            )
+            return
+
+        build_file = dist_info_dir / "BUILD"
+        build_file.write_text(
+            tomlkit.dumps(ctx.settings.get_package_settings(req.name))
+        )
+        if data_to_add:
+            build_file.write_text(tomlkit.dumps(data_to_add))
```

Comment:
Does this over-write the file, or append to it? Maybe what we really want to do is add the extra data from the plugin to the other settings as a key in the other dict. Something like `custom-metadata-extensions`? It's a long name, so something shorter that captures the meaning might be better.

Does TOML impose any restrictions on the syntax for keys? (no spaces, case-insensitive, etc.)

Reply:
okay I can either add the custom metadata in some key or another option is that i make the settings data as the default function. If a plugin is specified then all the data is added only from there?

TOML does have some restrictions: https://github.com/toml-lang/toml/blob/main/toml.md#keys

Reply:
How likely is it that data we are pulling from YAML or other places won't be compatible with TOML?

Reply:
i think the tomlkit handles it for us. It was able to successfully convert this yaml without an error:

```yaml
packages:
  stevedore:
    download_source:
      url: https://files.pythonhosted.org/packages/e7/c1/b210bf1071c96ecfcd24c2eeb4c828a2a24bf74b38af13896d02203b1eec/stevedore-${version}.tar.gz
      destination_filename: stevedore-custom-${version}.tar.gz
    resolver_dist:
      sdist_server_url: "https://pypi.org/simple"
      include_sdist: false
      include_wheels: true
    changelog:
      "1.0.0 hahaha":
        - test hello world
```